src/lib.rs

```diff
@@ -78,69 +80,73 @@ impl Agent {
 
 // ***** Income evolution *****
 
-pub fn income(agent: Agent, rate: f32) { // add rate to global parameters
-    let a = agent.age;
-    let i = agent.social_class.init_income;
-    let g = agent.social_class.growth;
-    let s = agent.social_class.start;
-    let r = agent.social_class.retire + 1;
-    if a < s {
-        0
-    } else if a < r {
-        i * g.pow(a - s) 
-    } else {
-        i * r
+impl Agent {
+    // TODO add rate to global parameters
+    pub fn income(&self, rate: f32) -> f32 {
```

Comment:
A few changes here:

1) Make income a method on the agent, like `fn up` above. It organizes the code better than having a bunch of free-standing functions
2) A method that does `fn blah(self, rate: f32)` or the original `fn income(agent: Agent, rate: f32)` will "consume" and destroy the agent, so nothing can use it afterwards. The `&` means immutable borrow. The income method doesn't need to modify the agent at all, it just needs a read-only version of it.   https://doc.rust-lang.org/rust-by-example/fn/methods.html
3) For most of the functions here, I think you want the return value. `-> f32` means this returns a float